packages/itwin/tree-widget/src/test/trees/models-tree/internal/ModelsTreeVisibilityHandler.test.ts

```diff
@@ -1763,6 +1763,178 @@ describe("HierarchyBasedVisibilityHandler", () => {
+          expectations: () => VisibilityExpectations.all("visible"),
+        },
+        {
+          name: "adjusts modeled elements display accordingly when subModel category display is changed",
+          nodeCreatorFunc: (ids) => createCategoryHierarchyNode(ids.modeledElementId, ids.categoryOfSubModelId, true),
+          expectations: (ids) => ({
+            subject: () => "partial",
+            model: () => "partial",
+            category: ({ categoryId, modelId }) => {
+              if (categoryId === ids.categoryOfSubModelId) {
+                return "visible";
+              }
+              if (modelId === ids.modeledElementId) {
+                return "hidden";
+              }
+              return "partial";
+            },
+            groupingNode: ({ elementIds }) => {
+              if (elementIds.includes(ids.modeledElementId)) {
+                return "partial";
+              }
+              if (elementIds.includes(ids.elementOfSubModelId1)) {
+                return "visible";
+              }
+              return "hidden";
+            },
+            element: ({ elementId }) => {
+              if (elementId === ids.elementOfSubModelId1) {
+                return "visible";
+              }
+              if (elementId === ids.elementOfSubModelId2) {
+                return "hidden";
+              }
+              return "partial";
+            },
+          }),
+        },
+        {
+          name: "adjusts modeled elements display accordingly when subModel categories element display is changed",
+          nodeCreatorFunc: (ids) =>
+            createElementHierarchyNode({ modelId: ids.modeledElementId, categoryId: ids.categoryOfSubModelId, elementId: ids.elementOfSubModelId1 }),
+          expectations: (ids) => ({
+            subject: () => "partial",
+            model: () => "partial",
+            category: ({ categoryId, modelId }) => {
+              if (categoryId === ids.categoryOfSubModelId) {
+                return "visible";
+              }
+              if (modelId === ids.modeledElementId) {
+                return "hidden";
+              }
+              return "partial";
+            },
+            groupingNode: ({ elementIds }) => {
+              if (elementIds.includes(ids.modeledElementId)) {
+                return "partial";
+              }
+              if (elementIds.includes(ids.elementOfSubModelId1)) {
+                return "visible";
+              }
+              return "hidden";
+            },
+            element: ({ elementId }) => {
+              if (elementId === ids.elementOfSubModelId1) {
+                return "visible";
+              }
+              if (elementId === ids.elementOfSubModelId2) {
+                return "hidden";
+              }
+              return "partial";
+            },
+          }),
+        },
+      ];
+      testCases.forEach(({ name, nodeCreatorFunc, expectations }) => {
```

Comment:
I'm not sure I like the idea of defining these tests like this.
- The single test iModel has more stuff than necessary for each test.
- When looking at test definitions I have to constantly scroll down to see what each ID means.
- The test definition names aren't very clear in describing the tested situation.

Reply:
Thanks, I think it's better after your changes.